include/exadg/solvers_and_preconditioners/solvers/iterative_solvers_dealii_wrapper.h

```diff
@@ -272,6 +277,11 @@ class SolverGMRES : public SolverBase<VectorType>
     }
     else
     {
+      if(not preconditioner.is_initialized())
+        preconditioner.initialize();
+      else if(update_preconditioner)
+        preconditioner.update();
```

Comment:
aims to make sure that preconditioner is not initialized and - immediately after this - updated again.